sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveLateralColumnAliasReference.scala

```diff
@@ -104,7 +112,10 @@ object ResolveLateralColumnAliasReference extends Rule[LogicalPlan] {
       // phase 2: unwrap
       plan.resolveOperatorsUpWithPruning(
         _.containsPattern(LATERAL_COLUMN_ALIAS_REFERENCE), ruleId) {
-        case p @ Project(projectList, child) if p.resolved
+        case p @ Project(projectList, child) if
+          (p.resolved
+            // can also work on nodes that contains Window but have all others resolved
+            || (projectList.exists(hasWindowExpression) && exprsAndChildResolved(p)))
```

Comment:
`p.resolved || (projectList.exists(hasWindowExpression) && exprsAndChildResolved(p))` shows up multiple times. Shall we abstract them into a method?